### PR TITLE
Support babel `data-type="module"`

### PR DESCRIPTION
**Pull request by fisker**
Ref: https://github.com/babel/babel/pull/11466

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/printer-html.js [hunk 1]
```diff
@@ -72,24 +72,25 @@ function embed(path, print, textToDoc, options) {
...
               breakParent,
               printOpeningTagPrefix(node, options),
-              stripTrailingHardline(
-                textToDoc(
-                  value,
-                  options.parser === "html" && parser === "babel"
-                    ? {
-                        parser,
-                        __babelSourceType:
-                          node.attrMap && node.attrMap.type === "module"
```
Comment (fisker) [inline src/language-html/printer-html.js:86]: This is mistake from #8173, should be `node.parent.attrMap `

**Comment by fisker** [comment]
@thorn0 About [this](https://github.com/prettier/prettier/pull/8173#discussion_r422095375), I tried to remove `opts &&` in  `const sourceType = opts && opts.__babelSourceType === "script" ? "script" : "module";` again, and found it's called from [here](https://github.com/prettier/prettier/blob/d420202df4443009d54a6288d65e5d3219aa746f/tests_integration/custom-parsers/custom-rename-parser.js#L4).

**Comment by thorn0** [comment]
A changelog entry is needed. Maybe just modify the one for #8173.

**Comment by fisker** [comment]
The `changelog_unreleased/html` dir contains `pr-8137.md` `pr-8173.md` `pr-8371.md`... What a coincidence!

**Comment by thorn0** [comment]
Yep, I noticed that too. :)

**Comment by fisker** [comment]
Done.

**Comment by fisker** [comment]
Another coincidence, when I see this babel PR yesterday, we both have 627 open issues. 😄

**Pull request merged**